frontend/test/metabase/visualizations/visualizations/Table.unit.spec.js

```diff
@@ -0,0 +1,92 @@
+import React, { useState } from "react";
+import { thaw } from "icepick";
+import userEvent from "@testing-library/user-event";
+import { renderWithProviders, screen, within } from "__support__/ui";
+
+import {
+  SAMPLE_DATABASE,
+  ORDERS,
+  metadata,
+} from "__support__/sample_database_fixture";
+import ChartSettings from "metabase/visualizations/components/ChartSettings";
+
+import Question from "metabase-lib/Question";
+
+const setup = () => {
+  const Container = () => {
+    const [question, setQuestion] = useState(
+      new Question(
+        {
+          dataset_query: {
+            type: "query",
+            query: {
+              "source-table": ORDERS.id,
+            },
+          },
+          database: SAMPLE_DATABASE.id,
+          visualization_settings: {},
+        },
+        metadata,
+      ),
+    );
+
+    const onChange = update => {
+      setQuestion(q => {
+        const newQuestion = q.updateSettings(update);
+        return new Question(thaw(newQuestion.card()), metadata);
+      });
+    };
+
+    return (
+      <ChartSettings
+        onChange={onChange}
+        series={[
+          {
+            card: question.card(),
+            data: {
+              rows: [],
+              cols: ORDERS.fields.map(f => f.column()),
+            },
+          },
+        ]}
+        initial={{ section: "Data" }}
+        noPreview
+        question={question}
+      />
+    );
+  };
+
+  renderWithProviders(<Container />);
+};
+
+describe("table settings", () => {
+  it("should show you related columns in structured queries", async () => {
+    setup();
+    expect(await screen.findByText("More columns")).toBeInTheDocument();
```

Comment:
Nit: I don't think you need to use `await` for the testing-library methods like `screen` and `within`, they don't return promises

Reply:
I think the `findBy` functions return promises, as they will re-query after a certain interval to give something time to show up. `getBy` and `queryBy` are synchronous, so they wouldn't require the awaits.